### PR TITLE
[WIP] Support for universal app

### PR DESCRIPTION
**Pull request by yuanyan**
*Before* submitting a pull request, please make sure the following is done...

1. Fork the repo and create your branch from `master`.
2. If you've added code that should be tested, add tests!
3. If you've changed APIs, update the documentation.
4. Ensure the test suite passes (`npm test`).
5. Make sure your code lints (`npm run lint`) - we've done our best to make sure these rules match our internal linting guidelines.

### CODE REVIEW

File: packages/driver-universal/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# driver-universal
+
+> Driver for Universal App.
+
+
+
+
+
```
Comment (wssgcg1213) [inline packages/driver-universal/README.md]: 文件末尾换行符留 1

File: packages/rax-cli/src/generator/universalapp/README.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+The page will reload if you make edits.
+
+### `npm run lint`
+
+You will see the lint errors in the console.
+
+### `npm run build`
+
+Builds the app for production to the `build` folder.
```
Comment (wssgcg1213) [inline packages/rax-cli/src/generator/universalapp/README.md]: 祈使句 Build

File: packages/rax-cli/src/generator/universalapp/README.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
+# WebApp Starter Kit
+
+## Getting Started
+
+### `npm run start`
+
+Runs the app in development mode.
```
Comment (wssgcg1213) [inline packages/rax-cli/src/generator/universalapp/README.md]: Run

File: packages/rax-cli/src/generator/universalapp/README.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
+# WebApp Starter Kit
+
+## Getting Started
+
+### `npm run start`
```
Comment (wssgcg1213) [inline packages/rax-cli/src/generator/universalapp/README.md]: `npm start` or `npm run start`, 哪个更好?

File: packages/rax-cli/src/generator/universalapp/pages/home/index.jsx [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+      name: 'MiniApp'
+    });
+  }
+
+  render() {
+    return (
+      <View className="app">
+        <Header>
+          <Image mode="widthFix" src="https://gw.alicdn.com/tfs/TB1omutPwHqK1RjSZFkXXX.WFXa-498-498.png" className="logo" />
+          <Text className="title" onTap={this.handleTap}>Welcome to {this.state.name}</Text>
```
Comment (wssgcg1213) [inline packages/rax-cli/src/generator/universalapp/pages/home/index.jsx]: `onTap` -> `onClick`

**Comment by wssgcg1213** [comment]
1. 补充了 driver-universal
2. 补充了 rax-scripts 的 build universal app 逻辑
3. universal-app 脚手架从 class component 改成 hooks 形式
4. 其它文档补充

File: packages/rax-cli/src/generator/universalapp/_eslintrc.js [hunk 6]
```diff
@@ -0,0 +1,176 @@
+module.exports = {
```
Comment (yacheng) [inline packages/rax-cli/src/generator/universalapp/_eslintrc.js]: lint 规则可以收拢 eslint-config-rax

File: packages/driver-universal/src/index.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import { isWeex, isWeb } from 'universal-env';
+import * as DriverDOM from 'driver-dom';
```
Comment (yacheng) [inline packages/driver-universal/src/index.js]: universal app 是否可以考虑将页面 750 的规则引入进来

File: packages/rax-cli/src/generator/universalapp/pages/home/index.css [hunk 8]
```diff
@@ -0,0 +1,18 @@
+page {
```
Comment (yacheng) [inline packages/rax-cli/src/generator/universalapp/pages/home/index.css]: 这里没有使用到

File: packages/driver-universal/src/index.js (hunk 7 above)
Comment (wssgcg1213) [reply packages/driver-universal/src/index.js]: DriverDOM 默认没有这个规则吗, 需要额外配置?

Comment (wssgcg1213) [reply packages/driver-universal/src/index.js]: weex 和 web 应该是需要的一致

File: packages/rax-cli/src/generator/universalapp/pages/home/index.css (hunk 8 above)
Comment (wssgcg1213) [reply packages/rax-cli/src/generator/universalapp/pages/home/index.css]: done

File: packages/driver-universal/src/index.js [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+import * as DriverWeex from 'driver-weex';
+
+let DriverUniversal;
+if (isWeex) {
+  DriverUniversal = DriverDOM;
```
Comment (yacheng) [inline packages/driver-universal/src/index.js]: 写反了

File: packages/driver-universal/src/index.js [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+} else if (isWeb) {
+  DriverUniversal = DriverWeex;
```
Comment (yacheng) [inline packages/driver-universal/src/index.js]: 写反了

File: packages/driver-universal/src/index.js (hunk 9 above)
Comment (wssgcg1213) [reply packages/driver-universal/src/index.js]: 已修复

**Comment by wssgcg1213** [comment]
我重新提一个 PR

**Pull request closed without merging**